src/cpp/communication/Track.cpp

```diff
@@ -58,18 +61,23 @@ Track::~Track()
     // Disable reader
     disable();
 
-    // Set exit status and call transmit thread to terminate
-    exit_.store(true);
-    available_data_condition_variable_.notify_all();
+    // Unset callback on the Reader (this is needed as Reader will live longer than Track)
+    reader_->unset_on_data_available_callback();
+
+    // Set exit status and call transmit thread to awake and terminate. Then wait for it.
+    exit_.store(true); // This is not needed to be guarded as it is atomic
+    transmit_condition_variable_.notify_all();
     transmit_thread_.join();
 }
 
-void Track::enable()
+void Track::enable() noexcept
 {
     std::lock_guard<std::recursive_mutex> lock(track_mutex_);
```

Comment:
Perhaps this mutex is not necessary.

Reply:
It is, as enable could be called without construction or destruction purposes.